tests/evaluate/test_default_evaluator.py

```diff
@@ -2990,6 +2983,174 @@ def test_default_metrics_as_custom_metrics_static_dataset():
     assert "exact_match/v1" in results.metrics.keys()
 
 
+def test_derived_metrics():
+    def metric_1(predictions, targets, metrics):
+        return MetricValue(
+            scores=[0, 1],
+            justifications=["first justification", "second justification"],
+            aggregate_results={"aggregate": 0.5},
+        )
+
+    def metric_2(predictions, targets, metrics, metric_1):
+        return MetricValue(
+            scores=[score * 5 for score in metric_1.scores],
+            justifications=[
+                "metric_2: " + justification for justification in metric_1.justifications
+            ],
+            aggregate_results={
+                **metric_1.aggregate_results,
+                **metrics["toxicity/v1"].aggregate_results,
+            },
+        )
+
+    def metric_3(predictions, targets, metric_1, metric_2):
+        return MetricValue(
+            scores=[score - 1 for score in metric_2.scores],
+            justifications=metric_1.justifications,
+            aggregate_results=metric_2.aggregate_results,
+        )
+
+    with mlflow.start_run():
+        data = pd.DataFrame(
+            {
+                "question": ["words random", "This is a sentence."],
+                "truth": ["words random", "This is a sentence."],
+                "answer": ["words random", "This is a sentence."],
+            }
+        )
+        results = evaluate(
+            data=data,
+            targets="truth",
+            predictions="answer",
+            model_type="text",
+            extra_metrics=[
+                make_metric(eval_fn=metric_1, greater_is_better=True, version="v1"),
+                make_metric(eval_fn=metric_2, greater_is_better=True, version="v2"),
+                make_metric(eval_fn=metric_3, greater_is_better=True),
```

Comment:
as i understand it we are not building a relationship between metric version and ordering. are there some corner cases that we should confirm -- ie. v2 of a metric that depends on v1 (idk if this is possible)

Reply:
Since it's not currently possible to define a dependency with a version (e.x. metric1 is dependent on metric2/v1), the only case I can think of here is that metric1 is dependent on metric2, but we provide two different versions of the same metric to extra_metrics() to be calculated. In this case metric1 will take the value of whichever one comes first in extra_metrics=[]. I can add a warning here if we detect two metrics with the same name but different versions.